api_tests/lib/ledgers/lnd_instance.ts

```diff
@@ -1,34 +1,47 @@
 import { ChildProcess, spawn } from "child_process";
-import { E2ETestActorConfig } from "../config";
-import { mkdirAsync, waitUntilFileExists, writeFileAsync } from "../utils";
+import { waitUntilFileExists, writeFileAsync } from "../utils";
 import * as path from "path";
-import { Logger } from "log4js";
 import getPort from "get-port";
 import { LogReader } from "./log_reader";
 import { Lnd } from "comit-sdk";
+import whereis from "@wcjiang/whereis";
+import { LightningInstance, LightningNodeConfig } from "./lightning";
+import { Logger } from "log4js";
 
-export class LndInstance {
+export class LndInstance implements LightningInstance {
     private process: ChildProcess;
-    private lndDir: string;
     public lnd: Lnd;
     private publicKey?: string;
 
-    constructor(
+    public static async new(
+        dataDir: string,
+        name: string,
+        logger: Logger,
+        bitcoindDataDir: string
+    ) {
+        return new LndInstance(
+            dataDir,
+            name,
+            logger,
+            bitcoindDataDir,
+            await getPort(),
+            await getPort()
+        );
+    }
+
+    private constructor(
+        private readonly dataDir: string,
+        private readonly name: string,
         private readonly logger: Logger,
-        private readonly testLogDir: string,
-        private readonly actorConfig: E2ETestActorConfig,
-        private readonly bitcoindDataDir: string
+        private readonly bitcoindDataDir: string,
+        private readonly lndP2pPort: number,
```

Comment:
Just choose between `p2p` and `lightning` please.

Reply:
Going for p2p for consistency with the bitcoin code.